src/main/java/org/junit/rules/Timeout.java

```diff
@@ -143,16 +143,17 @@ protected final boolean getLookingForStuckThread() {
      * @since 4.12
      */
     protected Statement createFailOnTimeoutStatement(
-            Statement statement) throws Exception {
+            Statement statement, Description description) throws Exception {
```

Comment:
> I'm in favour of modifying the methods signature and making it private. Users should create a new rule instead of extending Timeout. They could use the builder for creating the statement.

Agree on all points here!

> We could not add the new feature without breaking the extension point.

We can mark the old method as `@Deprecated`, and add a new, private method with a signature that takes a `Description`. The old method would work as before, not providing a `Description` to the `FailOnTimeout`--thus threads created via the old method would be given generic names like `"Time-limited test"`.